python/lsst/daf/butler/transfers/_context.py

```diff
@@ -155,14 +150,16 @@ def saveDatasets(self, refs: Iterable[DatasetRef], *,
                 exports = [rewrite(export) for export in exports]
             self._dataset_ids.add(ref.getCheckedId())
             assert ref.run is not None
-            datasets[ref.datasetType, ref.run].extend(exports)
+            self._datasets[ref.datasetType, ref.run].extend(exports)
         self.saveDataIds(dataIds, elements=elements)
-        for (datasetType, run), records in datasets.items():
-            self._backend.saveDatasets(datasetType, run, *records)
 
     def _finish(self) -> None:
         """Delegate to the backend to finish the export process.
 
         For use by `Butler.export` only.
         """
+        for element in self._registry.dimensions.sorted(self._records.keys()):
```

Comment:
Does this sorting help us out at all with DM-26324 ?

Reply:
It's a start, but we'd also need to sort the iterable provided on the line below, and that's harder, because we don't have a way to sort data IDs or dimension records (something I realized after my Jira post about trying it out).